### PR TITLE
Feat(eos_validate_state): ANTA Bump to ANTA 0.12.0 and update code

### PR DESCRIPTION
**Pull request by carl-baillargeon**
## Change Summary

Upgrade `eos_validate_state` ANTA requirements to 0.12.0. Update code + docstrings.

## Related Issue(s)

N/A

## Component(s) name

`arista.avd.eos_validate_state` **ANTA**

## How to test
` pip3 install anta==0.12.0`
`ansible-playbook playbooks/fabric-validate.yaml -e use_anta=true`

### User Checklist

- N/A

### Repository Checklist

- [X] My code has been rebased from devel before I start
- [X] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [X] My change requires a change to the documentation and documentation have been updated accordingly.
- [X] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md [hunk 1]
```diff
@@ -66,6 +69,14 @@ title: Ansible Collection Role eos_valudate_state - Preview Integration with ANT
   '
   ```
 
+  To upgrade the package:
+
+  ```shell
+  pip3 install -U '
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md]: I don't think you need -U when you assign the version with ==

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py [hunk 2]
```diff
@@ -29,110 +35,82 @@
...
     hostvars: Mapping,
     logging_level: str,
     skipped_tests: list[dict],
     ansible_tags: dict | None = None,
-    save_catalog_name: str | None = None,
-    dry_run: bool = False,
+    save_catalog_name: Path | None = None,
     yaml_dumper: Dumper | None = NoAliasDumper,
+    *,
+    dry_run: bool = False,
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py:50]: why are you forcing this to be a kwarg?

**Review by emilarista — APPROVED** [review]
Tested with DC and MPLS designs, no issues found. LGTM!

File: ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py (hunk 2 above)
Comment (carl-baillargeon) [reply ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py:50]: Calling a function with boolean positional arguments is confusing as the meaning of the boolean value is not clear to the caller.

Check `flake8-boolean-trap` for reference.

File: ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md (hunk 1 above)
Comment (carl-baillargeon) [reply ansible_collections/arista/avd/roles/eos_validate_state/ANTA-Preview.md]: Oh right I didn't know that! Thanks

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**